nodes/scene/timer.py

```diff
@@ -0,0 +1,574 @@
+                        self.timer_loop_count -= 1
+
+                else:  # not looping => stop ?
+                    # debug("start < NO looping")
+                    self.timer_time = 0
+                    self.timer_status = TIMER_STATUS_STOPPED
+                    self.last_timer_status = TIMER_STATUS_STARTED
+
+            elif self.timer_time > self.timer_duration:
+                if self.timer_loops:  # looping ?
+                    if self.timer_loop_count == self.timer_loops:  # at the last loop => expire
+                        # debug("start >= looping = N")
+                        self.timer_time = self.timer_duration
+                        self.timer_status = TIMER_STATUS_EXPIRED
+                        self.last_timer_status = TIMER_STATUS_STARTED
+
+                    else:  # not at the last loop => wrap around to next loop
+                        # debug("start >= looping < N")
+                        self.timer_time = self.timer_time % self.timer_duration
+                        self.timer_loop_count += 1
+
+                else:  # not looping => expire ?
+                    # debug("start >= NO looping")
+                    self.timer_time = self.timer_duration
+                    self.timer_status = TIMER_STATUS_EXPIRED
+                    self.last_timer_status = TIMER_STATUS_STARTED
+
+        elif self.timer_status == TIMER_STATUS_STOPPED and not sticky:
+            delta_time = (new_frame - old_frame) / fps * self.timer_speed
+            if delta_time > 0:
+                self.timer_status = self.last_timer_status
+
+        elif self.timer_status == TIMER_STATUS_EXPIRED and not sticky:
+            delta_time = (new_frame - old_frame) / fps * self.timer_speed
+            if delta_time < 0:
+                self.timer_status = self.last_timer_status
+
+        # keep the time within range
+        self.timer_time = max(0, min(self.timer_time, self.timer_duration))
+
+        # debug("Timer {0}: Status = {1}, Elapsed Time = {2}, Remaining Time = {3}, Expired = {4}".format(self.id(), self.status(), self.elapsed_time(), self.remaining_time(), self.expired()))
+
+
+class SvTimerOperatorCallback(bpy.types.Operator):
+    ''' Timer Operation (Start, Stop, Pause, Reset, Expire) '''
+    bl_idname = "nodes.sv_timer_callback"
+    bl_label = "Sv Ops Timer callback"
+
+    function_name: StringProperty()  # what function to call
+
+    def execute(self, context):
+        n = context.node
+        getattr(n, self.function_name)(context)
+        return {"FINISHED"}
+
+
+class SvTimerNode(bpy.types.Node, SverchCustomTreeNode, SvAnimatableNode):
+    """
+    Triggers: Timer, Elapse
+    Tooltip: Manage elapsed time via Start, Stop, Pause, Reset, Expire operations.
+    """
+    bl_idname = 'SvTimerNode'
+    bl_label = 'Timer'
+    bl_icon = 'PREVIEW_RANGE'
+
+    timers: CollectionProperty(name="Timers", type=SvTimerPropertyGroup)
```

Comment:
you'll want to have a way to rebuild this `timers` data from the serialized state, required by 
- gists...and
- presets..and 
- monads to some degree.

This will tell the export code to ignore that property (it's too complex to decompose automatically... at least for the current export code)
```diff
+  properties_to_skip_iojson = ["timers"]
```
see: https://github.com/nortikin/sverchok/search?q=properties_to_skip_iojson&unscoped_q=properties_to_skip_iojson

and implement `storage_get_data`, `storage_set_data`
https://github.com/nortikin/sverchok/blob/e3fc7a5dc900935d607bd8f3167ace1956f09c46/nodes/script/multi_exec.py#L251-L266

Reply:
look at a few nodes' implementation of `storage_get_data`, `storage_set_data`

Reply:
Thanks @zeffii . If you don’t mind for the time being I was going add the skip iojson to the node and plan to add the set/get storage feature for export in the next update. Some of the timer attributes are updated every update cycle from the node itself, but things like current state (started, paused etc) are not so export/import SV json trees will not restore these. 

Blender seems to preserve these settings (timer collection group settings) when saving/loading blend files. I wonder if SV import/export could somehow tap into blender’s mechanism to export these collection properties.

Reply:
it's easier to implement the storage_get/set stuff while the datastructure is still fresh in your mind.

i'll have a look.

Reply:
I’m sure it would be best to do it now, though i’m just afraid it may be a rabbit hole for me and I don’t have much time right now to tackle this, and I wanted to merge the PR for others to start playing with the node. 

Of all the timer properties.. 

# this is currently used for debugging
timer_id

# these are updated by the node with every update cycle (so i dont’ think they need to be stored)
timer_duration
timer_speed
timer_loops

# these would need to be stored
last_timer_status
timer_status
timer_time

timer_loop_count
timer_last_operation

timer_start_frame
timer_last_frame